src/main/java/org/jabref/logic/citationkeypattern/AbstractCitationKeyPattern.java

```diff
@@ -33,12 +32,19 @@ public static List<String> split(String bibtexKeyPattern) {
         // A holder for fields of the entry to be used for the key
         List<String> fieldList = new ArrayList<>();
 
-        // Before we do anything, we add the parameter to the ArrayLIst
+        // Before we do anything, we add the parameter to the ArrayList
         fieldList.add(bibtexKeyPattern);
 
-        StringTokenizer tok = new StringTokenizer(bibtexKeyPattern, "[]", true);
-        while (tok.hasMoreTokens()) {
-            fieldList.add(tok.nextToken());
+        // regex to split around a character
+        String WITH_DELIMITER = "((?<=%1$s)|(?=%1$s))";
+
+        // Split around [ or ]
+        String[] bibtexKeyPatteStrings = bibtexKeyPattern.split(
```

Comment:
Outdated - since we should keep StringTokenizer.

Typo. Please rename to `bibtexKeyPatterStrings`.

Didn't IntelliJ warn you?

![image](https://user-images.githubusercontent.com/1366654/229437020-cf12efc6-5efd-4abd-8097-6b2f011ae481.png)